src/control/drpc/drpc_client.go

```diff
@@ -56,6 +60,8 @@ func (c *ClientConnection) Connect() error {
 		return errors.Wrap(err, "dRPC connect")
 	}
 
+	c.connMu.Lock()
```

Comment:
I think you will want to wrap these critical sections around the whole function. Same is true for Close() below. Otherwise I think we could still get into some strange states even if we avoid the data race.

Reply:
could you elaborate please?

Reply:
I think it should be before the `IsConnected()` check, to avoid dialing twice and leaving one of the connections open forever. I know `IsConnected()` also uses the lock -- maybe you could add a non-exported version of that function to be used internally.